implementation/src/main/java/io/smallrye/graphql/bootstrap/schema/helper/CollectionHelper.java

```diff
@@ -40,20 +38,17 @@
      * @param type the collection class
      * @return the collection
      */
-    public Collection newCollection(Class type) {
-        if (type.equals(Collection.class) || type.equals(List.class)) {
-            return new ArrayList();
-        } else if (type.equals(Set.class)) {
-            return new HashSet();
-        } else {
-            try {
-                Constructor constructor = type.getConstructor();
-                constructor.setAccessible(true);
-                return (Collection) constructor.newInstance();
-            } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException ex) {
-                LOG.error("Can not create new collection of [" + type.getName() + "]", ex);
-                return new ArrayList(); // default ?
+    public static Collection<?> newCollection(Class<?> type) {
+        try {
+            return (Collection<?>) type.newInstance();
+        } catch (Exception ex) {
+            if (LOG.isDebugEnabled()) {
```

Comment:
I think we don't need this check. Building the string is a minor performance hit compared to building the exception.